domain/application/state/application_test.go

```diff
@@ -1000,6 +1000,35 @@ func (s *applicationStateSuite) TestGetUnitNamesOneNotFound(c *gc.C) {
 	c.Assert(err, jc.ErrorIs, applicationerrors.UnitNotFound)
 }
 
+func (s *applicationStateSuite) TestGetApplicationIDByUnitName(c *gc.C) {
+	u1 := application.InsertUnitArg{
+		UnitName: "foo/666",
+	}
+	expectedAppUUID := s.createApplication(c, "foo", life.Alive, u1)
+
+	obtainedAppUUID, err := s.state.GetApplicationIDByUnitName(context.Background(), u1.UnitName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(obtainedAppUUID, gc.Equals, expectedAppUUID)
+}
+
+func (s *applicationStateSuite) TestGetApplicationIDByUnitNameUnitNotFound(c *gc.C) {
+	_, err := s.state.GetApplicationIDByUnitName(context.Background(), "failme")
+	c.Assert(err, jc.ErrorIs, applicationerrors.UnitNotFound)
+}
+
+func (s *applicationStateSuite) TestGetCharmModifiedVersion(c *gc.C) {
+	appUUID := s.createApplication(c, "foo", life.Alive)
+
+	charmModifiedVersion, err := s.state.GetCharmModifiedVersion(context.Background(), appUUID)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(charmModifiedVersion, gc.Equals, 7)
```

Comment:
**suggestion:** I think test should be readable and understandable from the test itself and this is not.

If we test that charmModifiedVersion should be 7, it should be defined in the test itself, either through a transaction directly in the test, either through some kind of option stuff passed to createApplication;

```go
appUUID := s.createApplication(c, "foo", life.Alive, WithCharmModifiedVersion(7))
```

IMHO, silently setting to "some value" in createApplication and checking "some value" in the test code is too prone to be broken for no good reason.